third_party/libunwindstack/Unwinder.cpp

```diff
@@ -211,6 +239,16 @@ void Unwinder::Unwind(const std::vector<std::string>* initial_map_names_to_skip,
       // Once a frame is added, stop skipping frames.
       initial_map_names_to_skip = nullptr;
     }
+    if (map_info != nullptr && ShouldStop(functions_to_stop, map_info->name(), rel_pc)) {
+      if (frame != nullptr) {
+        if (!resolve_names_ ||
+            !object->GetFunctionName(step_pc, &frame->function_name, &frame->function_offset)) {
+          frame->function_name = "";
+          frame->function_offset = 0;
```

Comment:
I guess that this is the same code as below but I don't get it. Why do we need to unset name and offset? Were they set previously?

Reply:
Actually I haven't questioned this. Yes, this is the code from below, which does not get executed because we break.
If I understand the existing code correctly, at least function_offset might get set and afterwards GetFunctionName returns false. So you would get some strange result.